### ISSUE TITLE
Method overloading and concatenation

### ISSUE THREAD

**Issue by vorov2** [vorov2/dyalect#44]
**Dyalect version**
0.4.2

**Describe the bug**
The following works as expected:

```swift
12.2 + " + " + 0.4
```

It is evaluated to a string`"12.2 + 0.4"` (because of implicit coversion). Here the meaning of `+` is concatenation, not addition. However, in a code like so:

```swift
func Float +(other) {
    this + " + " + other
}
12.2 + 0.4
```

Runtime is crushed with "stack overflow" because concatenation is no longer chosen as a correct operation here.

**Comment by vorov2** [comment]
To be fixed in 0.4.3

**Comment by vorov2** [comment]
Fix is in "feature"
